spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/spring/HealthCheckServiceConfigurator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.spring;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.AnnotationAwareOrderComparator;
+import org.springframework.core.annotation.Order;
+
+import com.linecorp.armeria.server.healthcheck.HealthCheckService;
+import com.linecorp.armeria.server.healthcheck.HealthCheckServiceBuilder;
+
+/**
+ * Interface used to configure a {@link HealthCheckService} on the default armeria server.
+ */
+@FunctionalInterface
+public interface HealthCheckServiceConfigurator extends Ordered {
+
+    /**
+     * Configures the {@link HealthCheckService} using the specified {@link HealthCheckServiceBuilder}.
+     */
+    void configure(HealthCheckServiceBuilder healthCheckServiceBuilder);
+
+    /**
+     * Returns the evaluation order of this configurator. A user can specify the order with an {@link Order}
+     * annotation when defining a bean with a {@link Bean} annotation.
+     *
+     * <p>Note that the default value of the {@link Order} annotation is {@link Ordered#LOWEST_PRECEDENCE}
+     * which equals to {@link Integer#MAX_VALUE}, but it is overridden to {@code 0} by this default method.
+     *
+     * @see Ordered#LOWEST_PRECEDENCE
+     * @see Ordered#HIGHEST_PRECEDENCE
+     * @see AnnotationAwareOrderComparator
+     */
+    default int getOrder() {
+        return 0;
+    }
```

Comment:
Other Armeria configurators for Spring do not extend `Ordered`.
Do we need to apply `Ordered` to others?
Any thought? @heowc

Reply:
Hmm, what configurators don't extend? I referred `ArmeriaServerConfigurator` and `DocServiceConfigurator`.
On the contrary, although I added `Ordered`, do we really need to support `Ordered` to here...? Let me watch this discussion.

Reply:
> Hmm, what configurators don't extend?

Oops, I checked different [ArmeriaServerConfigurator](https://github.com/line/armeria/blob/fbe90f0f2a952578012a552c26b3f4f8bdb33c8c/dropwizard2/src/main/java/com/linecorp/armeria/dropwizard/ArmeriaServerConfigurator.java#L25) file that is used for Dropwizard. 😅
Please ignore this review.